scheduler/src/cook/mesos/constraints.clj

```diff
@@ -93,17 +99,24 @@
     (let [; Look at attribute and running jobs to determine if vm has gpus
           vm-has-gpus? (or (get vm-attributes "COOK_GPU?") ; Set when putting attributes in cache
                            (some (fn gpu-task? [req]
-                                   (job-needs-gpus? (:job req)))
+                                   (:needs-gpus? req))
```

Comment:
The req is a TaskRequest, coming from the TaskRequestAdaptor. It is the needs-gpus? argument on the record.